### DISCUSSION TITLE
Is it possible to put iframe in termsText?

### DISCUSSION THREAD

**Discussion by AlexBab11** [keycloakify/keycloakify#687] [Q&A]
Hi, i was wondering if its possible to put an iframe for termsText in terms and conditions page. Tried to do it in i18n.ts file, but it is not shown, though for example if i try to put some text there or link it works fine.

**Comment by garronej** [comment]
Hello @AlexBab11,  

Yes, for better of for worse this is the expected behavior. It's how it behave in the default Keycloak login UI.  

Thankfully however, you can change this in your theme very easily.  

Just run:  

```bash
npx keycloakify eject-page 
```

Select, `terms.ftl` then apply the code that is printed on the console.  

Next open the `src/login/pages/Terms.tsx` page and apply this diff:  

```diff
-           <div id="kc-terms-text">{msg("termsText")}</div>
+           <div id="kc-terms-text" dangerouslySetInnerHTML={{ __html: msgStr("termsText") }} />
```

It is safe to do so because the terms html code is a trusted string that you have defined.  

More info on the configuration of terms and conditions: https://docs.keycloakify.dev/terms-and-conditions  

Best
